src/process/message_processor.go

```diff
@@ -4,75 +4,102 @@ import (
 	"BGPAlert/analyze"
 	"BGPAlert/common"
 	"BGPAlert/config"
-	"errors"
 	"fmt"
 	"strconv"
 	"time"
 )
 
 // Constantly read messages from channel, build up windows with frequency maps, calling analysis on windows when full
-func ProcessBGPMessages(msgChannel chan common.BGPMessage, config *config.Configuration) error {
-	maximumBuckets, err := strconv.Atoi(config.WindowSize)
-	if err != nil {
-		return errors.New("couldn't get maxBuckets, " + err.Error())
+func ProcessBGPMessages(msgChannel chan common.Message, config *config.Configuration) error {
+	// Parse windowSize from config
+	var maximumBuckets int
+	// If windowSize can't be parsed to an int -> just default to 30
+	if windowSize, err := strconv.ParseInt(config.WindowSize, 10, 64); err != nil {
+		fmt.Println("Inputted config size is not a number, defaulting to 30")
+		maximumBuckets = 30
+	} else {
+		maximumBuckets = int(windowSize)
 	}
 
 	// Each bucket is 60s
 	parseDuration := 60 * time.Second
 	maximumTimespan := time.Duration(maximumBuckets*60) * time.Second
 
-	window := common.Window{
-		Filter:    "none",
-		BucketMap: make(map[time.Time][]common.BGPMessage),
-	}
+	// Stores windows that will be used for analysis
+	var windows []common.Window
```

Comment:
Rather than an array that you have to loop through, this could be a map I think. So the filter string is held inside the message, but also used as the key into the map. For example: https://go.dev/play/p/3nBWacvY9Hv